mcs/tools/gensources/gensources.cs

```diff
@@ -163,17 +162,27 @@ public static class Program {
             }
         }
 
-        TextWriter output;
-        if (useStdout)
-            output = Console.Out;
-        else
-            output = new StreamWriter (outFile);
+        int parts = files.Count == 0 ? 0 : files.Max (f => f.SplitNumber);
+        TextWriter[] outputs = new TextWriter[parts + 1];
 
-        using (output) {
-            foreach (var fileName in fileNames)
-                output.WriteLine (fileName);
+        for (int i = 0; i < parts + 1; i++) {
+            outputs[i] = useStdout ? Console.Out : new StreamWriter (i == 0 ? outFile : $"{outFile}.part{i}");
```

Comment:
It'd be nice if useStdout + more than one part was an error, otherwise I think this is going to output a given file name to stdout multiple times, which is both unexpected and something that will produce errors if the list with duplicates is fed into csc or msbuild

Reply:
Added an error check.